includes/core/classes/class-query.php

```diff
@@ -65,24 +65,32 @@ public function get_past_events( int $number = 5 ): \WP_Query {
 	 * @param string $event_list_type  Type of event list: upcoming or past.
 	 * @param int    $number           Maximum number of events.
 	 * @param array  $topics           Array of topic slugs.
+	 * @param array  $venues           Array of venue slugs.
 	 *
 	 * @return \WP_Query
 	 */
-	public function get_events_list( string $event_list_type = '', int $number = 5, array $topics = array() ): \WP_Query {
+	public function get_events_list( string $event_list_type = '', int $number = 5, array $topics = array(), array $venues = array() ): \WP_Query {
 		$args = array(
 			'post_type'       => Event::POST_TYPE,
 			'fields'          => 'ids',
 			'no_found_rows'   => true,
 			'posts_per_page'  => $number,
 			'gp_events_query' => $event_list_type,
 		);
-
-		if ( ! empty( $topics ) ) {
+		if ( ! empty( $venues ) || ! empty( $topics ) ) {
```

Comment:
Think you need 3 statements if venue is empty and topics isn't, if topics is empty and venue isn't, if topics and venue aren't empty.

Reply:
Can probably do this cleaner actually with building the `tax_query` array based on the what is empty or not. If that's confusing let me know and I can just refactor it after you get it working with 3 statements.